### PR TITLE
Redirect grafana's http to https by nginx container

### PR DESCRIPTION
**Pull request by hongkailiu**
https://github.com/kubernetes/test-infra/issues/12854

### CODE REVIEW

File: prow/monitoring/grafana_configmaps.yaml [hunk 1]
```diff
@@ -1,6 +1,36 @@
...
+        proxy_temp_path       /tmp/proxy_temp_path;
+        fastcgi_temp_path     /tmp/fastcgi_temp;
+        uwsgi_temp_path       /tmp/uwsgi_temp;
+        scgi_temp_path        /tmp/scgi_temp;
+
+        server {
+            listen 80 default_server;
+            server_name _;
+            if ($http_x_forwarded_proto != 'https') {
+                return 301 https://monitoring.prow.k8s.io;
```
Comment (cblecker) [inline prow/monitoring/grafana_configmaps.yaml]: This can be generic:
```suggestion
                return 301 https://$host$request_uri;
```

Comment (hongkailiu) [reply prow/monitoring/grafana_configmaps.yaml]: Sorry for the late reply, @cblecker 
Just came back from PTO.

**Comment by cblecker** [comment]
/lgtm
/hold

Holding as this will need to be deployed.

**Comment by hongkailiu** [comment]
> Holding as this will need to be deployed.

Thanks. We will deploy it next week.

**Pull request merged**